lib/filterx/func-sdata.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024 Szilard Parrag
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "func-sdata.h"
+#include "filterx/object-primitive.h"
+#include "filterx/filterx-eval.h"
+#include "filterx/object-string.h"
+
+#include <string.h>
+
+
+FilterXObject *
+filterx_simple_function_is_sdata_from_enterprise(FilterXExpr *s, GPtrArray *args)
+{
+  if (args == NULL || args->len != 1)
+    {
+      filterx_simple_function_argument_error(s, "Requires exactly one argument", FALSE);
+      return NULL;
+    }
+
+  FilterXObject *obj = filterx_typecast_string(NULL, args);
+  if (!obj)
+    return NULL;
+  const gchar *str_value = filterx_string_get_value(obj, NULL);
+  if (!str_value)
+    return NULL;
+
+  gboolean contains = FALSE;
+  FilterXEvalContext *context = filterx_eval_get_context();
+  LogMessage *msg = context->msgs[0];
+
+  for (guint8 i = 0; i < msg->num_sdata && !contains; i++)
+    {
+      const gchar *value = log_msg_get_value_name(msg->sdata[i], NULL);
+      gchar *res = strstr(value, str_value);
```

Comment:
I think we should add the "@" in front of the string value.

Reply:
Not sure about this as [RFC424](https://datatracker.ietf.org/doc/html/rfc5424#section-6.3.2) says that there are names without @:
>Names that do not contain an at-sign ("@", ABNF %d64) are reserved
      to be assigned by IETF Review as described in [BCP26](https://datatracker.ietf.org/doc/html/bcp26) [[RFC5226](https://datatracker.ietf.org/doc/html/rfc5226)].
      Currently, these are the names defined in [Section 7](https://datatracker.ietf.org/doc/html/rfc5424#section-7).

Reply:
Yeah, but the function is there to find sdata coming from a specific enterprise.

We may need a function eventually to check for a specific block, but then that should not be named is_sdata_from_enterprise()

Reply:
Yes, thank you, forgot about that part for a moment! :sweat_smile: